### PR TITLE
feat(versioning): add rpm versioning

### LINKED ISSUE #20465: Add RPM versioning
### What would you like Renovate to be able to do?

Renovate should be able to parse and compare RPM versions. RPM versioning scheme is used for RPM based Linux distributions like Red Hat Linux, Rocky, or Mariner.

Note that this is different from [RedHat versioning](https://github.com/renovatebot/renovate/blob/main/lib/modules/versioning/redhat/index.ts)

### If you have any ideas on how this should be implemented, please tell us here.

I've previously implemented RPM versioning in C# and should be able to write it again in TypeScript.

### Is this a feature you are interested in implementing yourself?

Yes

### PR DESCRIPTION
**Pull request by JamieMagee**
## Changes

Adds RPM versioning scheme

## Context

Closes #20465

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/versioning/rpm/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
+import rpm from '.';
```
Comment (HonkingGoose) [inline lib/modules/versioning/rpm/index.spec.ts:1]: I think you should create a `lib/modules/versioning/rpm/readme.md` file that explains what your `rpm` versioning does? For example, here's the Debian versioning readme:

https://github.com/renovatebot/renovate/blob/main/lib/modules/versioning/debian/readme.md

**Comment by lmilbaum** [comment]
Can I help with this PR?

**Comment by JamieMagee** [comment]
> Can I help with this PR?

If I recall correctly, it's pretty close. Just needs some test fixes.

I've been a bit preoccupied after moving, and haven't got my desk set up yet. Once I do, I'll take a look at this again.

**Comment by laundry-96** [comment]
Hey! Is there any reason why this MR was closed? I'm looking to use this feature so I don't mind driving this to completion

**Comment by rarkins** [comment]
Please feel free to run with it. @JamieMagee was not able to continue it

**Pull request closed without merging**